app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,28 @@
+class QueueItemsController < ApplicationController
+  before_action :require_user
+  def index
+    @queue_items = current_user.queue_items
+  end
+
+  def create
+    # check to see if already in queue
+    already_in_queue = current_user.queue_items.any? {|item| item.video_id == params[:video_id].to_i} 
+    queue_item = current_user.queue_items.new(video_id: params[:video_id], position: current_user.queue_items.size + 1)
+
+    if already_in_queue
+      flash['error'] = "Video already exists in your queue"
+    elsif (!already_in_queue) 
+      queue_item.save
+    else
```

Comment:
This `else` will never be hit. You covered all possible conditions (except raised errors) when you added conditions for `already_in_queue` and `!already_in_queue`.